### DISCUSSION TITLE
Bilingual strings

### DISCUSSION THREAD

**Discussion by saumier** [culturecreates/artsdata-planet-lavitrine#5] [Modelling discussions] [answered]
How to model multi-lingual strings ?

I propose using language tags as follows:

Example:
```
 "name": {
          "fr": "Édifice Wilder - Espace Danse",
          "en": "Wilder Building - Dance Space",
          "@none": "Wilder Building - Dance Space"
        }
```

**Comment by christianroy** [comment]
As per my discussion with @GenFab28, we will wait for other choices if you think there are better options, but if not, we are comfortable for the current example...

**Comment by saumier** [comment]
Another choice is to introduce 2 properties like title and titleInLanguage.

For example this @context used in TD ([Thing Description Ontology](https://www.w3.org/2019/wot/td#titleInLanguage))

> title of the TD element (Thing, interaction affordance or data schema) with language tag. By convention, a language tag must be added to the object of 'titleInLanguage' triples. Otherwise, use 'title'.

Example @context:
```
 "title": {
        "@id": "td:title",
        "@language": "en"
    },
 "titles": {
        "@id": "td:titleInLanguage",
        "@container": "@language"
    }
```

Example JSON-LD:
```
 "name": {
        "Wilder Building - Dance Space" 
  },
 "names": {
        "fr": "Édifice Wilder - Espace Danse",
        "en": "Wilder Building - Dance Space"
   }
```

I do not recommend this approach because it forces a default language and provides 2 different places in the JSON-LD for a developer to read the data depending on which one is available.

**Comment by GenFab28** [comment]
@saumier : merci Gregory. Si tu ne recommandes pas, on y va avec la première proposition.

**Comment by saumier** [comment]
See issue https://github.com/culturecreates/artsdata-planet-lavitrine/issues/13

**Comment by saumier** [comment]
@christianroy Note that I removed the "@" in-front of "@none" by adding "none": "@none" to the @context.  I also updated the discussion description.

**Comment by saumier** [comment]
@christianroy I was just in a W3C meetings and they recommend not aliasing "@none" as "none" because it is not a language and represents the absence of language. If you don't have any objections, I'll remove the alias so "@none" will appear when there is no @lanuage assigned to the literal.  See issues https://github.com/culturecreates/artsdata-planet-lavitrine/issues/15

**Reply by christianroy** [reply]
ok, pas de soucis!

**Comment by saumier** [comment, marked as answer]
All language strings have a language tag prefix like "en" or "fr". When there is no language defined at all, then the prefix  is "@none".

Example:
```
 "name": {
          "fr": "Édifice Wilder - Espace Danse",
          "en": "Wilder Building - Dance Space",
          "@none": "Wilder Building - Dance Space"
        }
```
